igamt-lite-controller/src/main/java/gov/nist/healthcare/tools/hl7/v2/igamt/lite/web/controller/SearchController.java

```diff
@@ -0,0 +1,115 @@
+		}
+		return datatypeService.findByScope(SCOPE.HL7STANDARD.name());
+	}
+	
+	@RequestMapping(value = "/datatype", method = RequestMethod.GET, produces = "application/json")
+	public Datatype getDatatype(@RequestParam(value="name") String name,@RequestParam(value="hl7Version") String hl7Version) throws DataNotFoundException {
+		if(name != null && !name.isEmpty() && hl7Version != null && !hl7Version.isEmpty()){
+			return datatypeService.findByNameAndVesionAndScope(name, hl7Version, SCOPE.HL7STANDARD.name());
+		}
+		return null;
+	}
+	
+	@RequestMapping(value = "/segments", method = RequestMethod.GET, produces = "application/json")
+	public List<Segment> getSegments(@RequestParam(value="name", required=false) String name) throws DataNotFoundException {
+		if(name != null && !name.isEmpty()){
+			return segmentService.findByNameAndScope(name, SCOPE.HL7STANDARD.name());
+		}
+		return segmentService.findByScope(SCOPE.HL7STANDARD.name());
+	}
+	
+	@RequestMapping(value = "/segment", method = RequestMethod.GET, produces = "application/json")
+	public Segment getSegment(@RequestParam(value="name") String name,@RequestParam(value="hl7Version") String hl7Version) throws DataNotFoundException {
+		if(name != null && !name.isEmpty() && hl7Version != null && !hl7Version.isEmpty()){
+			return segmentService.findByNameAndVersionAndScope(name, hl7Version, SCOPE.HL7STANDARD.name());
+		}
+		return null;
+	}
+	
+	@RequestMapping(value = "/messages", method = RequestMethod.GET, produces = "application/json")
+	public List<Message> getMessages(@RequestParam(value="name", required=false) String name) throws DataNotFoundException {
+		if(name != null && !name.isEmpty()){
+			return messageService.findByNameAndScope(name, SCOPE.HL7STANDARD.name());
+		}
+		return messageService.findByScope(SCOPE.HL7STANDARD.name());
+	}
+	
+	@RequestMapping(value = "/message", method = RequestMethod.GET, produces = "application/json")
+	public Message getMessage(@RequestParam(value="name") String name,@RequestParam(value="hl7Version") String hl7Version) throws DataNotFoundException {
+		if(name != null && !name.isEmpty() && hl7Version != null && !hl7Version.isEmpty()){
+			return messageService.findByNameAndVersionAndScope(name, hl7Version, SCOPE.HL7STANDARD.name());
+		}
+		return null;
+	}
+	
+	@RequestMapping(value = "/valueSets", method = RequestMethod.GET, produces = "application/json")
+	public List<Table> getValueSets(@RequestParam("scope") String scope, @RequestParam(value="bindingIdentifier", required=false) String bindingIdentifier) throws DataNotFoundException {
+		if(scope.equals(SCOPE.HL7STANDARD.name()) || scope.equals(SCOPE.PHINVADS.name())){
+			if(bindingIdentifier != null && !bindingIdentifier.isEmpty()){
+				return tableService.findByBindingIdentifierAndScope(bindingIdentifier, scope);
+			}
+			return tableService.findByScope(scope);
+		}
+		return tableService.findByScope(SCOPE.HL7STANDARD.name());
```

Comment:
if the scope does not match neither HL7STANDARD Nor PHINVADS We should return empty values.